src/freenet/client/filter/HTMLFilter.java

```diff
@@ -953,49 +953,57 @@ private static Map<String, TagVerifier> getAllowedTagVerifiers()
 					emptyStringArray));
 		String[] group2 =
 			{
-				"span",
-				"address",
-				"em",
-				"strong",
-				"dfn",
-				"code",
-				"samp",
-				"kbd",
-				"var",
-				"cite",
 				"abbr",
 				"acronym",
-				"sub",
-				"sup",
-				"dt",
-				"dd",
-				"tt",
-				"i",
+				"address",
+				"article",
+				"aside",
 				"b",
+				"bdi",
+				"bdo",
 				"big",
-				"small",
-				"strike",
-				"s",
-				"u",
-				"noframes",
-				"fieldset",
-// Delete <noscript> / </noscript>. So we can at least see the non-scripting code.
-//				"noscript",
-				"xmp",
-				"listing",
-				"plaintext",
 				"center",
-				"bdo",
-				"aside",
+				"cite",
+				"code",
+				"dd",
+				"details",
+				"dfn",
+				"dt",
+				"em",
+				"fieldset",
+				"figcaption",
+				"figure",
+				"footer",
 				"header",
+				"hgroup",
+				"i",
+				"kbd",
+				"listing",
+				"main",
+				"mark",
 				"nav",
-				"footer",
-				"article",
+				"noframes",
+				// Delete <noscript> / </noscript>. So we can at least see the non-scripting code.
+				//"noscript",
+				"plaintext",
+				"rp",
+				"rt",
+				"ruby",
+				"s",
+				"samp",
 				"section",
-				"hgroup",
-				"wbr",
+				"small",
+				"span",
+				"strike",
+				"strong",
+				"sub",
 				"summary",
-				"details"};
+				"sup",
+				"tt",
+				"u",
+				"var",
+				"wbr",
+				"xmp"};
```

Comment:
which elements did you add? the reordering makes that hard to figure out.

Reply:
"bdi","figcaption","figure","main","mark","rp","rt","ruby". I reordered them because it's hard to find which element is still missing. All these new elements are added to tests HTML5_TAGS and HTML5_BDI_RUBY.

Reply:
OK — thank you! Supporting ruby looks like a big improvement for Asian languages.